tests/qnn/test_keras.py

```diff
@@ -34,24 +34,44 @@ def model(get_circuit, n_qubits, output_dim):
     layer1 = KerasLayer(c, w, output_dim)
     layer2 = KerasLayer(c, w, output_dim)
 
-    model = tf.keras.models.Sequential(
-        [
-            tf.keras.layers.Dense(n_qubits),
-            layer1,
-            tf.keras.layers.Dense(n_qubits),
-            layer2,
-            tf.keras.layers.Dense(output_dim),
-        ]
-    )
+    if isinstance(output_dim, tuple):
+
+        model = tf.keras.models.Sequential()
+        model.add(tf.keras.layers.Dense(n_qubits))
+        model.add(layer1)
+        model.add(tf.keras.layers.Flatten())
+        model.add(tf.keras.layers.Dense(n_qubits))
+        model.add(layer2)
+        model.add(tf.keras.layers.Flatten())
```

Comment:
Do we need a layer after each circuit to take the absolute value, since the output of the circuit is complex?

Reply:
I don't think we need to take the absolute values since the calculations do take place even with the `complex128` type of tensor. But looking at it from a logical perspective (whether we need said complex numbers or only their real parts), I'm not sure about that. In any case, that `Lambda` layer for taking absolute values can be applied after our `KerasLayer` and before the `Flatten` layer. We will still need the `Flatten` layer since the `Lambda` layer with lambda function `lambda x: tf.abs(x)` does not change the shape of the outgoing dm, before we send it to the `Dense` layer.